lib/datadog/tracing/contrib/rack/header_tagging.rb

```diff
@@ -37,10 +37,16 @@ def self.tag_response_headers(span, headers, configuration)
                      whitelist = configuration[:headers][:response] || []
                      whitelist.each_with_object({}) do |header, result|
                        header_value = headers[header]
-                       unless header_value.nil?
-                         header_tag = Tracing::Metadata::Ext::HTTP::ResponseHeaders.to_tag(header)
-                         result[header_tag] = header_value
-                       end
+
+                       next if header_value.nil?
+
+                       header_tag = Tracing::Metadata::Ext::HTTP::ResponseHeaders.to_tag(header)
+
+                       result[header_tag] = if header_value.is_a? Array
```

Comment:
We should document that this is done to maintain the value format between Rack 2 and 3.